src/screens/FeedScreen/hooks/useViewPostTimeHook.ts

```diff
@@ -0,0 +1,38 @@
+import useFeedUtilsHook from './useFeedUtilsHook';
+import {SOURCE_FEED_TAB} from '../../../utils/constants';
+import {setTimer, setViewPostTimeIndex} from '../../../context/actions/feeds';
+import {viewTimePost} from '../../../service/post';
+
+const useViewPostTimeHook = (dispatch, timer, viewPostTimeIndex) => {
+  const {getCurrentPostViewed, isSamePostViewed} = useFeedUtilsHook();
+  const sendViewPostTimeWithFeeds = async (feedsParam = [], withResetTime = false) => {
+    const currentTime = new Date();
+    const diffTime = currentTime.getTime() - timer.getTime();
+    const id = (feedsParam?.[viewPostTimeIndex] as any)?.id;
+    if (!id) return;
+
+    viewTimePost(id, diffTime, SOURCE_FEED_TAB);
+    if (withResetTime) setTimer(new Date(), dispatch);
+  };
+
+  const updateViewPostTime = (momentumEvent) => {
+    setViewPostTimeIndex(getCurrentPostViewed(momentumEvent), dispatch);
+    setTimer(new Date(), dispatch);
+  };
+
+  const onWillSendViewPostTime = (event, feedParams) => {
+    if (isSamePostViewed(event, viewPostTimeIndex)) return;
+    sendViewPostTimeWithFeeds(feedParams);
+    updateViewPostTime(event);
+  };
+
+  return {
+    getCurrentPostViewed,
+    sendViewPostTimeWithFeeds,
+    updateViewPostTime,
+    isSamePostViewed,
+    onWillSendViewPostTime
+  };
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `useViewPostTimeHook` function seems to be well implemented. However, there is a potential issue with the `sendViewPostTimeWithFeeds` function. It's an asynchronous function but it's not being awaited when called in `onWillSendViewPostTime`. This could lead to unexpected behavior if the promise returned by `sendViewPostTimeWithFeeds` rejects or takes longer than expected to resolve.

```diff
-    sendViewPostTimeWithFeeds(feedParams);
+    await sendViewPostTimeWithFeeds(feedParams);
```
Please note that you'll need to make `onWillSendViewPostTime` an async function to use `await`.

Also, I noticed that `viewTimePost` function call at line 14 doesn't handle any potential errors. It would be good to wrap this in a try-catch block to handle any exceptions that might occur during the execution of the `viewTimePost` function.

```diff
-    viewTimePost(id, diffTime, SOURCE_FEED_TAB);
+    try {
+      await viewTimePost(id, diffTime, SOURCE_FEED_TAB);
+    } catch (error) {
+      console.error('Error sending view time post:', error);
+    }
```

Remember to add `await` before `viewTimePost` as it's likely an asynchronous operation.


<!-- This is an auto-generated comment by OSS CodeRabbit -->